x/stablestake/keeper/msg_server_bond.go

```diff
@@ -17,7 +18,14 @@ func (k msgServer) Bond(goCtx context.Context, msg *types.MsgBond) (*types.MsgBo
 
 	params := k.GetParams(ctx)
 	sender := sdk.MustAccAddressFromBech32(msg.Creator)
-	depositCoin := sdk.NewCoin(params.DepositDenom, msg.Amount)
+
+	entry, found := k.assetProfileKeeper.GetEntry(ctx, params.DepositDenom)
+	if !found {
+		return nil, errors.New("invalid denom")
```

Comment:
Can you register error and use registered error?

Reply:
Okay no problem.